### PR TITLE
[Doc] Add documentation for FP8 W8A8

### PR DESCRIPTION
**Pull request by mgoin**
Initial documentation for the FP8 W8A8 quantization feature, detailing how to produce quantized checkpoints models ahead of time using AutoFP8 in addition to the dynamic quantization within vLLM.

### CODE REVIEW

File: docs/source/quantization/fp8.rst [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+You can use AutoFP8 to produce checkpoints with their weights quantized to FP8 ahead of time and let vLLM handle calculating dynamic scales for the activations at runtime for maximum accuracy. You can enable this with the `activation_scheme="dynamic"` argument.
+
+.. code-block:: python
+
+    from auto_fp8 import AutoFP8ForCausalLM, BaseQuantizeConfig
+
+    pretrained_model_dir = "meta-llama/Meta-Llama-3-8B-Instruct"
+    quantized_model_dir = "Meta-Llama-3-8B-Instruct-FP8-Dynamic"
+
+    examples = []
```
Comment (comaniac) [inline docs/source/quantization/fp8.rst:57]: We should have a comment saying this quantization doesn't require a calibration dataset.

File: docs/source/quantization/fp8.rst [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    git clone https://github.com/neuralmagic/AutoFP8.git
+    pip install -e AutoFP8
+
+This package introduces the `AutoFP8ForCausalLM` and `BaseQuantizeConfig` objects for managing how your model will be compressed.
+
+Offline Quantization with Dynamic Activation Scaling Factors
+------------------------------------------------------------
+
+You can use AutoFP8 to produce checkpoints with their weights quantized to FP8 ahead of time and let vLLM handle calculating dynamic scales for the activations at runtime for maximum accuracy. You can enable this with the `activation_scheme="dynamic"` argument.
```
Comment (comaniac) [inline docs/source/quantization/fp8.rst:48]: We should also emphasize that although this mode doesn't give you better performance, it reduces memory footprint compared to online quantization.

**Review by comaniac — APPROVED** [review]
Overall LGTM. Once we have decided where to put FP8 checkpoints we can add links to this documents too.

File: docs/source/quantization/fp8.rst (hunk 1 above)
Comment (mgoin) [reply docs/source/quantization/fp8.rst:57]: Yes, adding!

File: docs/source/quantization/fp8.rst (hunk 2 above)
Comment (mgoin) [reply docs/source/quantization/fp8.rst:48]: Yes good call out.

**Review by robertgshaw2-redhat — APPROVED** [review]
Approved. Can you please add a note that AutoFP8 is in early beta and subject to change

**Comment by mgoin** [comment]
> AutoFP8 is in early beta and subject to change

Sure, I'll add it to the repo itself.

**Pull request merged**